trace_replay/trace_replay.cc

```diff
@@ -575,13 +578,18 @@ Status Replayer::Replay() {
         return Status::Corruption("Invalid Column Family ID.");
       }
 
+      Iterator* single_iter = nullptr;
       if (iter_payload.cf_id == 0) {
         single_iter = db_->NewIterator(roptions);
       } else {
         single_iter = db_->NewIterator(roptions, cf_map_[iter_payload.cf_id]);
       }
       single_iter->Seek(iter_payload.iter_key);
-      ops++;
+      s = single_iter->status();
+      if (!s.ok()) {
+        delete single_iter;
+        break;
+      }
       delete single_iter;
```

Comment:
You have a delete in the if and the else (outside the break) statement.  Might as well move the delete before the if.

Reply:
Done.